spec/controllers/vms_controller_spec.rb

```diff
@@ -226,6 +226,33 @@ def set_old_path
         expect(response).to redirect_to(vms_path)
       end
     end
+
+    context 'when the vm requires more resouces than any host can provide' do
+      before do
+        summary_double = double
+        allow(summary_double).to receive_message_chain(:config, :numCpu).and_return(999)
+
+        allow(vm2).to receive(:summary).and_return(summary_double)
+        allow(vm2).to receive(:change_power_state)        
+        
+        post :change_power_state, params: { id: vm2.name }
+      end
+
+      it 'raises an error' do 
+        expect {
+          post :change_power_state, params: { id: vm2.name }
+        }.to raise_error(RbVmomi::Fault)
```

Comment:
Where would this exception be thrown when the API is mocked? Maybe you can try to figure out a way to get the api request to fail only for this test :)